lib/publish.js

```diff
@@ -20,30 +24,30 @@ const npmPublish = options => {
 		args.push('--otp', options.otp);
 	}
 
-	return execa('npm', args);
+	return execa(pkgManager, args);
 };
 
-const handleError = (task, err, options, message) => {
-	if (err.stderr.includes('one-time pass')) {
+const handleError = (err, pkgManager, task, options, input, message) => {
+	if (err.stderr.includes('one-time pass') || err.message.includes('user TTY')) {
```

Comment:
`user TTY` feels a bit too generic. What is the full output? Any better parts of the output we could test?

Reply:
The error message is “Can't answer a question unless a user TTY”. I had it check for the whole message at first but then I thought the message was oddly-phrased and liable to change.

Reply:
Another option might be to add —non-interactive, just add the —otp flag every time, and see if yarn ignores it if it’s not needed.